apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/SnifferConfigInitializer.java

```diff
@@ -202,6 +206,37 @@ private static void overrideConfigBySystemProp() {
         }
     }
 
+    /**
+     * set agent version(Described in MANIFEST.MF)
+     *
+     * */
+    private static void setAgentVersion() {
+        try {
+            ClassLoader classLoader = SnifferConfigInitializer.class.getClassLoader();
+            Enumeration<URL> resources = SnifferConfigInitializer.class.getClassLoader().getResources(JarFile.MANIFEST_NAME);
+            while (resources.hasMoreElements()) {
+                URL url = resources.nextElement();
+                LOGGER.info("SnifferConfigInitializer url:{}", url.toString());
+                try (InputStream is = url.openStream()) {
+                    if (is != null) {
+                        Manifest manifest = new Manifest(is);
+                        Attributes mainAttribs = manifest.getMainAttributes();
+                        String projectName = mainAttribs.getValue("Implementation-Vendor-Id");
+                        if (projectName != null) {
+                            if ("org.apache.skywalking".equals(projectName)) {
+                                String version = mainAttribs.getValue("Implementation-Version");
+                                AGENT_SETTINGS.put("agent.version", version);
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
I think you missed to set `Config.Agent.VERSION` here?

Reply:
yes, i will fix it!